cmd/gateway/s3x/gateway-s3x-ledger-multipart.go

```diff
@@ -56,9 +60,14 @@ func (ls *ledgerStore) PutObjectPart(bucketName, objectName, partHash, multipart
 	if err != nil {
 		return err
 	}
-	m.ObjectParts[partNumber] = ObjectPartInfo{
-		Number:   partNumber,
-		DataHash: partHash,
+	pn := int64(pi.PartNumber)
+	m.ObjectParts[pn] = ObjectPartInfo{
```

Comment:
We should have a check here:

```Go
if m.ObjectParts == nil {
    m.ObjectParts = make(map[int64]ObjectPartInfo)
}
```

This will prevent an edge case panic

Reply:
This is taken care of when m is constructed.

Reply:
If we pull `m` from IPFS that has nothing in the `ObjectParts` field it will cause a panic i believe.

Reply:
Good call